### PR TITLE
parser: support show keys from table from database syntax.

### PR DESCRIPTION
**Pull request by coocood**
Fixes issue https://github.com/pingcap/tidb/issues/2303

### CODE REVIEW

**Comment by coocood** [comment]
@tiancaiamao @zimulala PTAL

File: executor/show_test.go [hunk 1]
```diff
@@ -121,6 +121,10 @@ func (s *testSuite) TestShow(c *C) {
 	for i, r := range row {
 		c.Check(r, Equals, expectedRow[i])
 	}
+
+	tk.MustExec("use show_test_DB")
+	result = tk.MustQuery("SHOW index from show_index from test")
```
Comment (zimulala) [inline executor/show_test.go]: Add test about `like`.

**Comment by zimulala** [comment]
LGTM

**Comment by tiancaiamao** [comment]
LGTM

**Pull request merged**